dm/dm/worker/subtask.go

```diff
@@ -552,7 +552,10 @@ func (st *SubTask) markResultCanceled() bool {
 func (st *SubTask) Result() *pb.ProcessResult {
 	st.RLock()
 	defer st.RUnlock()
-	return st.result
+	tempProcessResult, _ := st.result.Marshal()
```

Comment:
It'd better check the error just for the sake of security.

Reply:
Got it. I have modified the code and added the error handling.